docs/01.CONFIGURATION.md

```diff
@@ -0,0 +1,249 @@
+  Enabling this feature provides detailed error messages where available, like line information for Syntax errors, variable names for Reference errors, Type/Range error messages for built-in routines, etc.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_ERROR_MESSAGES=0/1`                 |
+  | CMake:  | `--DJERRY_ERROR_MESSAGES=ON/OFF`             |
+  | Python: | `--error-messages=ON/OFF`                    |
+
+* ### Logging
+
+  This option can be used to enable log messages during runtime. When enabled the engine will use the `jerry_port_log` port API function to print relevant log messages.
+  This feature is disabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_LOGGING=0/1`                        |
+  | CMake:  | `-DJERRY_LOGGING=ON/OFF`                     |
+  | Python: | `--logging=ON/OFF`                           |
+
+* ### LCache
+
+  This option enables the LCache, allowing faster access to object properties. See [Internals](04.INTERNALS.md) for further details.
+  This option is enabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_LCACHE=0/1`                         |
+  | CMake:  | `<none>`                                     |
+  | Python: | `<none>`                                     |
+
+* ### Property hashmaps
+
+  This option enables the creation of hashmaps for object properties. See [Internals](04.INTERNALS.md) for further details.
+  This option is enabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_PROPRETY_HASHMAP=0/1`               |
+  | CMake:  | `<none>`                                     |
+  | Python: | `<none>`                                     |
+
+* ### Memory statistics
+
+  This option can be used to provide memory usage statistics either upon engine termination, or during runtime using the `jerry_get_memory_stats` jerry API function.
+  The feature can create a significant performance overhead, and should only be used for measurement purposes. This option is disabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_MEM_STATS=0/1`                      |
+  | CMake:  | `-DJERRY_MEM_STATS=ON/OFF`                   |
+  | Python: | `--mem-stats=ON/OFF`                         |
+
+* ### Heap size
+
+  This option can be used to adjust the size of the internal heap, represented in kilobytes. The provided value should be an integer, with a maximum value of 512.
```

Comment:
> maximum value of 512.

This is not true. It can be bigger and the cpointer32 will be enabled for >512 values.